lisa/energy_model.py

```diff
@@ -793,6 +793,55 @@ def _find_core_groups(cls, target):
         # Again, not strictly necessary, just more pleasant.
         return sorted(ret, key=lambda x: x[0])
 
+    def _simple_em_root(target, pd_attr, cpu_to_pd):
+        # pd_attr is a dict tree like this
+        # {
+        #   "pd0": {
+        #       "capacity": [236, 301, 367, 406, 446 ],
+        #       "frequency": [ 450000, 575000, 700000, 775000, 850000 ],
+        #       "power": [ 42, 58, 79, 97, 119 ]
+        #   },
+        #   "pd1": {
+        #       "capacity": [ 418, 581, 744, 884, 1024 ],
+        #       "frequency": [ 450000, 625000, 800000, 950000, 1100000 ],
+        #       "power": [ 160, 239, 343, 454, 583 ]
+        #   }
+        # }
+        def simple_read_idle_states(cpu, target):
+            # idle states are not supported in the simple model
+            # record 0 power for them all, but name them according to target
+            names = [s.name for s in target.cpuidle.get_states(cpu)]
+            return OrderedDict((name, 0) for name in names)
+
+        def simple_read_active_states(pd):
+            cstates = list(zip(pd['capacity'], pd['power']))
+            active_states = [ActiveState(c, p) for c, p in cstates]
+            return OrderedDict(list(zip(pd['frequency'], active_states)))
```

Comment:
no need for `list` since we iterate only once over the iterator returned by `zip` (and there is no risk it happens in the future since that iterator is "anonymous")

Reply:
Right. I just copied the existing code but that's a good opportunity to clean that up while at it :wink: